### PR TITLE
[feature] add anchor icon to heading anchors

### PR DESCRIPTION
**Pull request by sophiabrandt**
This addresses the feature request: https://github.com/panr/hugo-theme-terminal/issues/87#issuecomment-575556699

Hugo already creates heading anchors for title headers (h1, etc.).
This pull requests adds some CSS and a new class to show an icon next to a heading.
Now, when you hover over a heading, the CSS will show an anchor icon, which you can click to jump directly to the heading anchor.
Solution see here: 
https://discourse.gohugo.io/t/adding-anchor-next-to-headers/1726/14

You can see the feature in action here: https://www.rockyourcode.com/step-by-step-guide-to-manage-ocaml-installations-with-asdf/#install-asdf (hover over the heading)

Perhaps there are some ways to improve this solution, as I'm not familiar with Golang/Hugo nor that good with CSS.

### CODE REVIEW

File: layouts/_default/single.html [hunk 1]
```diff
@@ -28,7 +28,11 @@ <h1 class="post-title">
   {{ end }}
 
   <div class="post-content">
-    {{ .Content }}
+  {{- with .Content -}}
+      <div>
+        {{ . | replaceRE "(<h[1-9] id=\"([^\"]+)\".+)(</h[1-9]+>)" `${1}<a href="#${2}" class="hanchor" ariaLabel="Anchor"> 🔗&#xFE0E;</a> ${3}` | safeHTML }}
```
Comment (panr) [inline layouts/_default/single.html]:
```suggestion
        {{ . | replaceRE "(<h[1-9] id=\"([^\"]+)\".+)(</h[1-9]+>)" `${1}<a href="#${2}" class="hanchor" ariaLabel="Anchor"> &#8983;</a> ${3}` | safeHTML }}
```

Comment (panr) [reply layouts/_default/single.html]: I'd replace anchor icon with "Viewdata Square" https://www.compart.com/en/unicode/U+2317

File: layouts/_default/single.html [hunk 2]
```diff
@@ -28,7 +28,11 @@ <h1 class="post-title">
   {{ end }}
 
   <div class="post-content">
-    {{ .Content }}
+  {{- with .Content -}}
```
Comment (panr) [inline layouts/_default/single.html]: Nitpicking but could you fix the indention here? :D

File: source/css/post.css [hunk 3]
```diff
@@ -98,3 +98,13 @@
     }
   }
 }
+
+.hanchor {
+  font-size: 100%;
+  visibility: hidden;
+  color:silver;
```
Comment (panr) [inline source/css/post.css]: I'd replace this with:

```
  color: color-mod(var(--accent) alpha(90%));
  text-decoration: none;
  margin-left: 10px;
  visibility: hidden;
```

**Comment by panr** [comment]
Also, could you pull latest code changes? Some conflicts here 😬

File: layouts/_default/single.html (hunk 2 above)
Comment (sophiabrandt) [reply layouts/_default/single.html]: Of course. Sorry, my bad.

**Comment by sophiabrandt** [comment]
@panr 
Ugh, it's the first time I'm using these features (review, request, etc.). I hope I've done everything correctly. Can you please take a look?
I have no clue on how to get rid of the "Changes requested" status.
Everything should be ok, and I rebuild the assets, just to be sure.

**Comment by panr** [comment]
> Ugh, it's the first time I'm using these features (review, request, etc.). I hope I've done everything correctly. Can you please take a look?

@sophiabrandt you've done everything right 👍  Even marked conversations as resolved <3

> I have no clue on how to get rid of the "Changes requested" status.

Only reviewer (hello there!) can get rid of this status by marking it as `accepted` ✅ 

> Many thanks for this theme, and the work you're putting in. I appreciate it.

You're welcome, I do my best 💪

**Pull request merged**